cr.js

```diff
@@ -35,26 +46,78 @@ PriorityQueue.prototype.get_all_priorities = function() {
 
 // iterates through all the queue elements in priority-then-FIFO order
 PriorityQueue.prototype.forEach = function(callback) {
-    var keys = Object.keys(this.store).sort();
 
-    for (var a = keys.length; a > 0; a--) {
-        for (var b = 0; b < this.store[a].length; b++)
-            callback(this.store[a][b]);
+    for (var key in this.store) {
+        this.store[key].forEach(function(val, indexOfVal){
+            callback(key, val, indexOfVal)
+        });
```

Comment:
in the use of variable a, we were always assuming that priorities are always a complete range of numbers from 1 to x. What if we have priorities [1,2,4], and in the for loop iteration, a becomes 3, this.store[a][b] would be undefined